cs/src/Contracts/Tunnel.cs

```diff
@@ -37,9 +37,9 @@ public Tunnel()
     /// Gets or sets the generated ID of the tunnel, unique within the cluster.
     /// </summary>
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    [RegularExpression(TunnelIdPattern)]
-    [StringLength(TunnelIdLength, MinimumLength = TunnelIdLength)]
-    public string? TunnelId { get; set; }
+    [RegularExpression(OldTunnelIdPattern)]
+    [StringLength(OldTunnelIdLength, MinimumLength = OldTunnelIdLength)]
+    public virtual string? TunnelId { get; set; }
```

Comment:
Is a reason this is made `virtual`?

Reply:
Originally when we were going to share objects it had to be but it is not needed anymore so I removed it